v0/destinations/vero/transform.js

```diff
@@ -0,0 +1,142 @@
+const set = require("set-value");
+const get = require("get-value");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  getValueFromMessage,
+  isDefinedAndNotNull
+} = require("../../util");
+const { EventType } = require("../../../constants");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG, BASE_URL } = require("./config");
+
+// This function handles any tag add or removal requests.
+
+const tagResponseBuilder = async (message, tags, destination) => {
+  const response = defaultRequestConfig();
+
+  const payload = {};
+  const authToken = get(destination, "Config.authToken");
+  set(payload, "auth_token", authToken);
+
+  const id = getValueFromMessage(message, "userId");
+  set(payload, "id", id);
+
+  // add[] and remove[] arrays contain the tags to be added or removed.
+  const addTags = getValueFromMessage(tags, "add");
+  if (Array.isArray(addTags) && addTags.length > 0)
+    set(payload, "add", addTags);
+
+  const removeTags = getValueFromMessage(tags, "remove");
+  if (Array.isArray(removeTags) && removeTags.length > 0)
+    set(payload, "remove", removeTags);
+
```

Comment:
We should consider the scenario where no tags are there in the payload, then we don't need to make the call to tag endpoint.

Reply:
This check is happening in line 82 before calling this tagResponseBuilder function.